### PR TITLE
[AAP-49666] test broken data on infrastructure summary worksheet

### PR DESCRIPTION
**Pull request by ShaiahWren**
[AAP-49666]

## Description

 ✅ All test requirements covered:
  - Null/empty device type data - Tests handle None and empty string values gracefully
  - Malformed/corrupted input data - Tests malformed JSON and non-JSON data
  - Timezone edge cases - Tests different timezone formats and edge times
  - UUID collision scenarios - Tests duplicate host names with different remote IDs
  - Device type mapping edge cases - Tests non-string types, lists, nested objects, and very long
  names
  - Error handling - Comprehensive error handling across all scenarios
  - User-friendly error messages - Tests verify meaningful messages like "No indirect nodes found"

## Testing

uv run pytest metrics_utility/test/ccspv_reports/test_infrastructure_summary_broken_data.py

### Steps to Test

1.
2.
3.

## Required Actions

- [ ] Requires documentation updates

- [ ] Requires downstream repository changes

- [ ] Requires infrastructure/deployment changes

- [ ] Requires coordination with other teams

- [ ] Blocked by PR/MR: #XXX

### Code Quality

- [ ] Code is properly linted and formatted.
- [ ] All tests (existing and new) pass successfully.
- [ ] Tests are added or updated as needed.
- [ ] Code includes relevant comments for complex sections.
- [ ] Changes are reviewed and approved by at least two team members.
- [ ] Documentation is updated where applicable.  
       - If updates are required in the [handbook](https://github.com/ansible/handbook), create a separate PR for the handbook repo.

## Notes for Reviewers

Add any additional context or instructions for reviewers here - for example screenshots if needed.

### CODE REVIEW

File: metrics_utility/test/ccspv_reports/test_infrastructure_summary_broken_data.py [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+        mock_ws = self.create_mock_worksheet()
+
+        # Test handling of duplicate host names with different remote IDs
+        try:
+            current_row = report._build_data_section_infrastructure_summary(1, mock_ws, broken_df)
+            assert current_row is not None
+            assert mock_ws.cell.called
+
+            # Verify that both records are processed despite having the same host name
```
Comment (himdel) [inline metrics_utility/test/ccspv_reports/test_infrastructure_summary_broken_data.py]: shouldn't this also test *how* they were processed?

Should the host end up being counted as both vmware and kubernetes and appear in two sheets?

**Review by himdel — COMMENTED** [review]
All those...

```
try:
    ...
except Exception as e:
    pytest.fail(f'... {str(e)}')
```

seem to just hide the reason for the failure.

If the code throws an exception, the test will already fail, but without that try..except, it would also tell us where it failed.

**Comment by himdel** [comment]
There doesn't seem to be any test validating that the hosts were actually counted in some category.

Am I missing something?

I would expect a check that the generated table has a value of `1` (or `2`) for the right sheet-row-column (searched by the name of the expected category), and `0` in the other counts. (In all the tests, or most.)

**Review by himdel — APPROVED** [review]
LGTM :)
